ExampleMod/Common/GlobalItems/TorchExtractinatorGlobalItem.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.GlobalItems
+{
+	// These 2 files show off making a new extractinator type. This example converts any torch placing item into any other torch placing item. ModSystem.PostSetupContent is used instead of GlobalItem.SetStaticDefaults to determine which items are torch items because it needs to run after all mods setup their content.
+	public class TorchExtractinatorGlobalItem : GlobalItem
+	{
+		public override void ExtractinatorUse(int extractType, int extractinatorBlockType, ref int resultType, ref int resultStack) {
+			// If the extractinator type isn't torch, we won't change anything
+			if(extractType != ItemID.Torch)
+				return;
+
+			// If it is, we set stack to 1 and return a random torch. If the user is using the Chlorophyte Extractinator, we return Ultrabright Torch 10% of the time.
+			resultStack = 1;
+			if (extractinatorBlockType == TileID.ChlorophyteExtractinator && Main.rand.NextBool(10)) {
+				resultType = ItemID.UltrabrightTorch;
+				return;
+			}
+
+			resultType = Main.rand.Next(TorchExtractinatorModSystem.TorchItems);
```

Comment:
I believe from discussion (not from actually looking) that the chlorophyte extractinator _cycles_ through item pools, rather than randomly rerolling.